src/Telemetry/AzureDevOpsTimeline/Program.cs

```diff
@@ -41,7 +44,37 @@ private static void Main()
                                 o.Interval = c["Interval"];
                                 o.BuildBatchSize = c["BuildBatchSize"];
                             });
+
+                            services.AddSingleton<IAzureDevOpsClient>(p =>
+                            {
+                                IConfiguration c = p.GetRequiredService<IConfiguration>();
+
+                                if (!int.TryParse(c["ParallelRequests"], out int parallelRequests) || parallelRequests < 1)
+                                {
+                                    parallelRequests = 5;
+                                }
+
+                                return new AzureDevOpsClient(
```

Comment:
Why not restructure it to be more DI friendly? So that you can just "Configure" some options, and let it handle the construction?

Reply:
(Same comment applied below)

Reply:
The AzureDevOpsClient is used in other projects in this solution. I wanted to avoid touching those. And... it seems like the better thing to do with this class is replace it with the Azure DevOps Client SDK libraries.

Agree with the second client, though. Originally I was being fast since this is the only project that needs it, but that's not very friendly. So I'll improve it.

Reply:
I improved injection of `ITimelineTelemetryRepository`. I went down the road of improving `AzureDevOpsClient` but didn't like how it affects the other project.